browser/ts/index.ts

```diff
@@ -0,0 +1,370 @@
+     */
+    onError(handler: (err: Error) => void) {
+        this._errHandler = handler;
+        if (this._wrapper != undefined) {
+            this._wrapper.onError(this._errHandler);
+        }
+    }
+
+    /**
+     * Registers a handler which will be called for any incoming order events.
+     * Order events are fired whenver an order is added, canceled, expired, or
+     * filled. In order to ensure no events are missed, this should be called
+     * before startAsync.
+     *
+     * @param   handler                The handler to be called.
+     */
+    onOrderEvents(handler: (events: Array<OrderEvent>) => void) {
+        this._orderEventsHandler = orderEventsHandlerToMeshOrderEventsHandler(handler);
+        if (this._wrapper != undefined) {
+            this._wrapper.onOrderEvents(this._orderEventsHandler);
+        }
+    }
+
+    /**
+     * Starts the Mesh node in the background. Mesh will automatically find
+     * peers in the network and begin receiving orders from them.
+     */
+    async startAsync(): Promise<void> {
+        await this._waitForLoadAsync();
+        this._wrapper = await zeroExMesh.newWrapperAsync(this._config);
+        if (this._orderEventsHandler != undefined) {
+            this._wrapper.onOrderEvents(this._orderEventsHandler);
+        }
+        if (this._errHandler != undefined) {
+            this._wrapper.onError(this._errHandler);
+        }
+        return this._wrapper.startAsync();
+    }
+
+    /**
+     * Validates and adds the given orders to Mesh. If an order is successfully
+     * added, Mesh will share it with any peers in the network and start
+     * watching it for changes (e.g. filled, canceled, expired). The returned
+     * promise will only be rejected if there was an error validating or adding
+     * the order; it will not be rejected for any invalid orders (check
+     * results.rejected instead).
+     *
+     * @param   orders                An array of orders to add.
+     * @returns Validation results for the given orders, indicating which orders
+     * were accepted and which were rejected.
+     */
+    async addOrdersAsync(orders: Array<SignedOrder>): Promise<ValidationResults> {
+        await this._waitForLoadAsync();
+        if (this._wrapper == undefined) {
+            // If this is called after startAsync, this._wrapper is always
+            // defined. This check is here just in case and satisfies the
+            // compiler.
+            return Promise.reject(new Error('Mesh is still loading. Try again soon.'));
+        }
+        const meshOrders = orders.map(signedOrderToMeshSignedOrder);
+        const meshResults = await this._wrapper.addOrdersAsync(meshOrders);
+        return meshValidationResultsToValidationResults(meshResults);
+    }
+}
+
+function sleepAsync(ms: number): Promise<void> {
+    return new Promise(resolve => setTimeout(resolve, ms));
+}
+
+function meshSignedOrderToSignedOrder(meshSignedOrder: MeshSignedOrder): SignedOrder {
```

Comment:
Why you're calling `MeshSignedOrder` we call `StringifiedSignedOrder` in `mesh-rpc-client`. Thoughts on standardizing?

Reply:
Hmm... `StringifiedSignedOrder` is an okay name in this case because all the types are strings. However, I want to pick a name that highlights the pattern of converting between Go types and TypeScript types. I'm thinking that there will be other type conversions in the future and not just `string`  to `BigNumber`. I can't think of a good name to cover the use cases in both `mesh-rpc-client` and `mesh-browser`. Maybe something like `BasicSignedOrder`?

Reply:
My vote would perhaps be `StringifiedSignedOrder` for now, and if there are other types involved in the future, changing it then. Don't feel too strongly about this though, so we can also leave it for now.

Reply:
I'm going to go with `WrapperSignedOrder`, `WrapperOrderEvent`, `WrapperValidationResults`, etc. to more closely match the fact that these are types exposed by `MeshWrapper` that might need further conversion. This is what I was originally trying to show with the `Mesh*` prefix but I feel that name was too confusing. Happy to re-evaluate later.